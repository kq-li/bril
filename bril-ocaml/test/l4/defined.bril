# ARGS: defined
@main(cond: bool) {
.block1:
  a: int = const 47;
  b: int = const 42;
  br cond .left .right;
.left:
  b: int = const 1;
  c: int = const 5;
  jmp .end;
.right:
  a: int = const 2;
  e: int = const 10;
.end:
  d: int = sub a c;
  print d;
  ret;
}
