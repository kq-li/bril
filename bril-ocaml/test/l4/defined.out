current calculation: defined variables
ins:
	label_block1 -> {cond}
	label_left -> {a, b, cond}
	label_right -> {a, b, cond}
	label_end -> {a, b, c, cond, e}
outs:
	label_block1 -> {a, b, cond}
	label_left -> {a, b, c, cond}
	label_right -> {a, b, cond, e}
	label_end -> {a, b, c, cond, d, e}
